logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -170,28 +173,30 @@ def initialize(queue, size, wait)
         @size = size
         @wait = wait
 
-        @originals = Hash.new
-
         # TODO: disabled for https://github.com/elastic/logstash/issues/6055 - will have to properly refactor
         # @cancelled = Hash.new
 
+        #Sizing HashSet to size/load_factor to ensure no rehashing
+        @originals = java.util.HashSet.new(size * 4 / 3 + 1, 0.75)
```

Comment:
If we want to avoid rehashing ,why not set the load factor to 1? I agree that it makes sense to have a larger than needed hash table to avoid collisions for quantities this small, but if the goal is to ensure no rehashing, why keep it at 0.75?